cinema/views.py

```diff
@@ -1 +1,55 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession
+from cinema.serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    CinemaHallSerializer,
+    MovieSerializer,
+    MovieSessionSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieSessionDetailSerializer,
+    MovieSessionListSerializer,
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.prefetch_related("actors", "genres")
+
+    def get_serializer_class(self):
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+
+        if self.action == "list":
+            return MovieListSerializer
+
+        return MovieSerializer
```

Comment:
`return self.serializer_class` is better according to best practices. Fix everywhere